resolwe/flow/views/mixins.py

```diff
@@ -44,27 +22,21 @@ class ResolweCreateModelMixin(mixins.CreateModelMixin):
 
     """
 
-    def create(self, request, *args, **kwargs):
-        """Create a resource."""
-        ds_query = request.data.get('descriptor_schema', None)
-        if ds_query:
-            ds_query = get_descriptor_schames(ds_query, request.user)
-            try:
-                request.data['descriptor_schema'] = ds_query.latest().pk
-            except DescriptorSchema.DoesNotExist:
-                return Response(
-                    {'descriptor_schema': [
-                        'Invalid descriptor_schema slug "{}" - object does not exist.'.format(ds_query)]},
-                    status=status.HTTP_400_BAD_REQUEST)
-
+    def define_contributor(self, request):
+        """Define contibutor by adding it to request.data."""
         if request.user.is_anonymous:
             request.data['contributor'] = get_anonymous_user().pk
         else:
             request.data['contributor'] = request.user.pk
 
+        return request
```

Comment:
You don't need to return modified object as original object (passed to the function) is also changed.